File: frontend/src/api/booking.ts
import type { ResponseSuccess, ResponseSuccessWithData, ResponseWitMetaType } from "@/core/models/base";
import { privateHttp } from "./http";
import type { BookingByIdType, BookingViewType, CreateBookingType, UpdateBookingType } from "@/core/models/booking";

export async function getBooking(userId: number): Promise<ResponseWitMetaType<BookingViewType[]>> {
    return privateHttp.get(`/bookings/${userId}`)
}

export async function createBooking(payload: CreateBookingType): Promise<ResponseSuccess> {
    return privateHttp.post("/booking/create", payload)
}

export async function getBookingById(id: number): Promise<ResponseSuccessWithData<BookingByIdType>> {
    return privateHttp.get(`/booking/${id}`)
}

export async function deleteBooking(id: number): Promise<ResponseSuccess> {
    return privateHttp.delete(`/booking/${id}`)
}

export async function updateBooking(payload: UpdateBookingType): Promise<ResponseSuccessWithData<UpdateBookingType>> {
    return privateHttp.put("/booking/update", payload)
}

File: backend/internal/routes/booking.go
package routes

import (
	"chairTime/constant"
	"chairTime/internal/app"
	bookingController "chairTime/internal/controllers/booking"

	"github.com/labstack/echo/v4"
)

func BookingRoute(app *app.Application, groupRoute echo.Group) {
	groupRoute.GET("/booking/:id", app.RouteHandler(bookingController.GetBookingById), app.Authorization([]int{constant.UserRoleId}))
	groupRoute.GET("/bookings/:user_id", app.RouteHandler(bookingController.GetBookings), app.Authorization([]int{constant.UserRoleId}))
	groupRoute.PUT("/booking/update", app.RouteHandler(bookingController.UpdateBooking), app.Authorization([]int{constant.UserRoleId}))
	groupRoute.POST("/booking/create", app.RouteHandler(bookingController.CreateBooking), app.Authorization([]int{constant.UserRoleId}))
	groupRoute.DELETE("/booking/:bookingId", app.RouteHandler(bookingController.DeleteBooking), app.Authorization([]int{constant.UserRoleId}))
}
